packages/examples/parameters/allParamTypes.js

```diff
@@ -0,0 +1,44 @@
+/*
+// title       : All Parameters Demo
+// author      : Z3 Dev, Simon Clark
+// license     : MIT License
+// description : Example of interactive parameters
+// file        : allParamTypes.js
+// tags        : parameters, select, choice, checkbox, color, picker, slider, date, input, parameter
+*/
+
+const { primitives } = require('@jscad/modeling')
+
+const values = [3, 4, 5, 6]
+const captions = ['three', 'four', 'five', 'six']
+
+const getParameterDefinitions = () => {
+  return [
+    { name: 'group1', type: 'group', caption: 'Group 1: Text Entry' },
+    { name: 'text', type: 'text', initial: '', size: 20, maxLength: 20, caption: 'Plain Text:', placeholder: '20 characters' },
+    { name: 'int', type: 'int', initial: 20, min: 1, max: 100, step: 1, caption: 'Integer:' },
+    { name: 'number', type: 'number', initial: 2.0, min: 1.0, max: 10.0, step: 0.1, caption: 'Number:' },
+    { name: 'date', type: 'date', initial: '2020-01-01', min: '2020-01-01', max: '2030-12-31', caption: 'Date:', placeholder: 'YYYY-MM-DD' },
+    { name: 'email', type: 'email', initial: 'me@example.com', caption: 'Email:' },
+    { name: 'url', type: 'url', initial: 'www.z3d.jp', size: 40, maxLength: 40, caption: 'Url:', placeholder: '40 characters' },
+    { name: 'password', type: 'password', initial: '', caption: 'Password:' },
+
+    { name: 'group2', type: 'group', caption: 'Group 2: Interactive Controls' },
+    { name: 'checkbox', type: 'checkbox', checked: true, initial: '20', caption: 'Checkbox:' },
+    { name: 'color', type: 'color', initial: '#FFB431', caption: 'Color:' },
+    { name: 'slider', type: 'slider', initial: 3, min: 1, max: 10, step: 1, caption: 'Slider:' },
+    { name: 'choice1', type: 'choice', caption: 'Dropdown Menu:', values: [0, 1, 2, 3], captions: ['No', 'Yes', 'Maybe', 'So so'], initial: 2 },
+    { name: 'choice3', type: 'choice', caption: 'Dropdown Menu:', values: ['No', 'Yes', 'Maybe', 'So so'], initial: 2 },
+    { name: 'choice2', type: 'radio', caption: 'Radio Buttons:', values, captions, initial: 5 },
+
+    { name: 'group3', type: 'group', initial: 'closed', caption: 'Group 3: Initially Closed Group' },
+    { name: 'checkbox2', type: 'checkbox', checked: true, initial: '20', caption: 'Optional Checkbox:' }
+  ]
+}
+
+const main = (params) => {
+  console.log(params)
```

Comment:
not sure console should be used.

maybe a comment on what each of the params will contain would be more appropriate.

Reply:
What do you think should be done?  The console is a valuable tool in working with jscad scripts in the browser.

Reply:
Not sure actually. The example only provides examples of the differ types of parameters. What’s the goal? Examples? User documentation?

Maybe another example with working parameters would be better.